webpack/src/Extends/Host/PuppetTab/SubTabs/YamlTab/index.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import PropTypes from 'prop-types';
+
+import Skeleton from 'react-loading-skeleton';
+import EmptyState from 'foremanReact/components/common/EmptyState/EmptyStatePattern';
+import { STATUS } from 'foremanReact/constants';
+import { EmptyStateIcon } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { global_danger_color_200 as dangerColor } from '@patternfly/react-tokens';
+import { translate as __ } from 'foremanReact/common/I18n';
+import './YamlTab.scss';
+
+const YamlTab = ({ hostName }) => {
+  const params = { params: { name: hostName, format: 'yml' } };
+  const url = `${window.location.origin.toString()}/foreman_puppet/hosts/${hostName}/externalNodes`;
+  const { response: yaml, status } = useAPI('get', url, params);
+
+  if (status === STATUS.PENDING) {
+    return <Skeleton count={5} />;
+  }
+
+  if (status === STATUS.ERROR) {
+    const icon = (
+      <EmptyStateIcon icon={ExclamationCircleIcon} color={dangerColor.value} />
+    );
+    return (
+      <EmptyState
+        header={__('Error!')}
+        icon={icon}
+        description={yaml?.response?.data?.error?.message}
+      />
+    );
+  }
+  if (yaml !== '' || yaml !== undefined) {
+    return (
+      <div className="yaml-tab">
+        <textarea value={yaml} readOnly />
```

Comment:
@marisvirik should we use here a [code-block](https://www.patternfly.org/v4/components/code-block) component instead of the HTML textarea ?

Reply:
Looks like there is only textarea available in foreman_ansible. This can still be fixed in a follow-up PR, I think.

Reply:
There is also the ['react-json-tree'](https://github.com/theforeman/foreman_ansible/blob/c31e984a66ee2bf6a74756a981abc5c4dced108b/webpack/components/ReportJsonViewer.js#L13) component in Foreman_Ansible if that's what you meant @MariSvirik, though that's not for our yaml code:
![Screenshot (91)](https://user-images.githubusercontent.com/26363699/171616373-1bf6b740-f6a1-4595-a879-73edeef005b0.png)

I would suggest to use the [code-block](https://www.patternfly.org/v4/components/code-block) component.

Reply:
Changed the textarea to use `code-block`